client/src/main/java/io/pravega/client/stream/impl/PositionImpl.java

```diff
@@ -55,37 +62,42 @@ public PositionImpl(Map<SegmentWithRange, Long> segments) {
     }
     
     @Builder(builderClassName = "PositionBuilder")
-    PositionImpl(Map<Segment, Long> ownedSegments, Map<Segment, Range> segmentRanges) {
-        this.ownedSegments = ownedSegments;
+    PositionImpl(Map<Segment, Long> ownedSegments, Map<Segment, Range> segmentRanges, List<Entry<Segment, Long>> updatesToSegmentOffsets) {
```

Comment:
The semantics of which of these collections are copied or not should be spelled out in a javadoc. It would avoid the bug below.

Reply:
Yes, I definitely can add javadoc explaining the main purpose of this builder and the way it is currently used. But I prefer to defer this task until we agree the final way to do this (whether passing `updatesToSegmentOffsets` is valid or not).

Reply:
I have added javadoc to clarify the goal of the builder method.